$NetBSD: patch-19778,v 1.1 2017/11/15 03:31:56 manu Exp $

From 3bfb4e454eb639f9d1a370dc056eb5b2ef38e484 Mon Sep 17 00:00:00 2001
From: Emmanuel Dreyfus <manu@netbsd.org>
Date: Wed, 15 Nov 2017 04:04:39 +0100
Subject: [PATCH] Disable gfid2path by default on NetBSD

NetBSD storage of extended attributes for UFS1 badly scales when
the list of extended attributes names rises. gfid2path can add as
many extended attributes names as we have files, hence we keep it
disabled for performance sake.

> Change-Id: Id77b5f5ceb4d5eba1b3362b4b9fc693450ffbc2b
> Signed-off-by: Emmanuel Dreyfus <manu@netbsd.org>
> BUG: 1129939

Change-Id: I5232a0bd8cb30fa42124f49b1054153f9c31f108
Signed-off-by: Emmanuel Dreyfus <manu@netbsd.org>
BUG: 1513258
---
 xlators/storage/posix/src/posix.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/xlators/storage/posix/src/posix.c b/xlators/storage/posix/src/posix.c
index d858878d3..577515263 100644
--- xlators/storage/posix/src/posix.c
+++ xlators/storage/posix/src/posix.c
@@ -7967,7 +7967,18 @@ struct volume_options options[] = {
         },
         { .key = {"gfid2path"},
           .type = GF_OPTION_TYPE_BOOL,
+#ifdef __NetBSD__
+          /* 
+	   * NetBSD storage of extended attributes for UFS1 badly 
+	   * scales when the list of extended attributes names rises.
+	   * This option can add as many extended attributes names
+	   * as we have files, hence we keep it disabled for performance
+	   * sake.
+	   */
+          .default_value = "off",
+#else
           .default_value = "on",
+#endif
           .description = "Enable logging metadata for gfid to path conversion"
         },
         { .key = {"gfid2path-separator"},
-- 
2.11.0

